abi/src/specs.rs

```diff
@@ -795,3 +797,12 @@ impl MessageParamSpecBuilder {
         self.spec
     }
 }
+
+fn serialize_selector<S>(selector: &u32, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let b = selector.to_be_bytes();
+    let hex = format!("0x{:02X}{:02X}{:02X}{:02X}", b[0], b[1], b[2], b[3]);
+    serializer.serialize_str(&hex)
+}
```

Comment:
Please also provide a simple test case for this conversion. (Will be super simple since we are going to start from `[u8; 4]` according to the other review.